### ISSUE TITLE
sortable doesn't work with array data

### ISSUE THREAD

**Issue by davismj** [aurelia-ui-toolkits/aurelia-kendoui-bridge#716]
Sortable works great for static html lists. Not so great for data-driven lists. Take the example from the demo, but actually using Aurelia:

app.js
```javascript
export class BasicUse {

  tracks = [
    { title: 'Papercut', duration: '3:04' },
    { title: 'One Step Closer', duration: '2:35' },
    { title: 'With You', duration: '3:23' },
    { title: 'Points of Authority', duration: '3:20' },
    { title: 'Crawling', duration: '3:29' },
    { title: 'Runaway', duration: '3:03' },
    { title: 'By Myself', duration: '3:09' },
    { title: 'In the End', duration: '3:36' },
    { title: 'A Place for My Head', duration: '3:04' },
    { title: 'Forgotten', duration: '3:14' },
    { title: 'Cure for the Itch', duration: '2:37' },
    { title: 'Pushing Me Away', duration: '3:11' }
  ]

  hint(e) {
    return $(e).clone().addClass('sortable-basic-use-hint');
  }

  placeholder(e) {
    return $(e).clone().addClass('sortable-basic-use-placeholder').text('drop here');
  }
  
  change() {
    console.log(this.tracks.map((t) => t.title));
  }
}
```

app.html
```html
<template>
  <require from="./basic-use.css!css"></require>
  <require from="aurelia-kendoui-bridge/sortable/sortable"></require>
  
  <div id="example">
    <div id="playlist">
      <div id="playlist-title"><span>My Playlist</span></div>
      <ul id="sortable-basic"
          ak-sortable="k-cursor-offset.bind: {top: -10, left: -230}; 
                                k-hint.call: hint($event); 
                                k-placeholder.call: placeholder($event);"
          k-on-change.delegate="change()">
        <li class="sortable" repeat.for="track of tracks">${track.title}<span>${track.duration</span></li>
      </ul>
    </div>
  </div>
  </template>
```

The output is of course the original array, unsorted, which doesn't reflect the change in the sortable. So, let's add a function to make the change:

app.js
```javascript
    change({ detail: { oldIndex, newIndex }}) {
      let [item] = this.tracks.splice(oldIndex, 1);
      this.tracks.splice(newIndex, 0, item);
  }
```

Now the data model is correct, but the sortable list goes nuts and everything is out of order.

**Comment by JeroenVinke** [comment]
Good point. Created a gist so I can mess with it later: https://gist.run/?id=861f98e06711e42caf4f1a40b9c2a22d

**Comment by davismj** [comment]
My guess is that you have this fixed already. Sortable has an `end` event that is default preventable: http://docs.telerik.com/kendo-ui/api/javascript/ui/sortable#events-end. By preventing default, you can manually take over.

However, default prevention isn't being propagated back to Kendo. You've made this fix but have yet to release it: https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/commit/39298e34e2a7703f8cf9406c10e6cbc5d2c5a1d7.

While we're discussing, it seems that the events that a queued on the microtask queue are not default preventable either: https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/blob/master/src/common/widget-base.js#L284. It might make sense to pull the prevent default logic right into the util.fireKendoEvent method.

**Comment by JeroenVinke** [comment]
> My guess is that you have this fixed already. Sortable has an end event that is default preventable: http://docs.telerik.com/kendo-ui/api/javascript/ui/sortable#events-end. By preventing default, you can manually take over.

Now that is a nice event to have 😄 

> However, default prevention isn't being propagated back to Kendo. You've made this fix but have yet to release it: 39298e3.

Ah, my bad. Will do this soon. 

> While we're discussing, it seems that the events that a queued on the microtask queue are not default preventable either: https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/blob/master/src/common/widget-base.js#L284. It might make sense to pull the prevent default logic right into the util.fireKendoEvent method.

in https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/issues/691 someone had the same issue but with the `filtering` event instead, which is put on the queue as well. In the end it worked for him though, so it should work I think

**Comment by JeroenVinke** [comment]
Released that patch as 1.2.1
